front/src/applications/editor/Editor.tsx

```diff
@@ -185,6 +195,81 @@ const Editor = () => {
     }
   }, []);
 
+  /**
+   * When the component mounts
+   * => get the searchParams
+   * => if theere is a selection param, select the entities and focus on them
+   */
+  useEffect(() => {
+    if (urlInfra) {
+      const params = searchParams.get('selection');
+      if (!params && searchParams.size !== 0) {
+        dispatch(
+          setFailure({
+            name: t('translation:Editor.tools.select-items.errors.unable-to-select'),
+            message: t('translation:Editor.tools.select-items.errors.invalid-url'),
+          })
+        );
+        navigate(`/editor/${urlInfra}`);
+      }
+      const paramsList = params?.split('|');
+      if (paramsList && paramsList.length === 1) {
```

Comment:
Add a if loop to evacuate the case where paramsList is undefined or empty, and to factorize, directly get the selected entities from the paramList. You can then manage the case where there is only 1 object or the one where there are several. You can also move the try/catch to factorize code
```ts
      if (paramsList && paramsList.length) {
        const selectedEntities = paramsList.map((param) => {
          const [objType, entityId] = param.split('~');
          return {
            id: entityId,
            type: objType as EditoastType,
          };
        });
        try {
          if (selectedEntities.length === 1) { ... }
          else { ... }
        } catch { ... }
```

Reply:
Done